### PR TITLE
POC-13

### PR DESCRIPTION
**Pull request by P-1-X-3-L**

### CODE REVIEW

File: src/Components/Login/Login.test.jsx [hunk 1]
```diff
@@ -1,9 +1,47 @@
...
+        isLoggedIn: true,
+        userName: 'test',
+      },
+    }));
+    const { container } = render(<Provider store={mockstore}><Login /></Provider>);
     expect(container).toMatchSnapshot();
   });
+  test('Login Failed', () => {
+    // userEvent.type(screen.getAllByRole('textbox')[0], '');
+    // userEvent.type(screen.getAllByRole('textbox')[1], '');
```
Comment (pranavakkian) [inline src/Components/Login/Login.test.jsx]: Avoid comments
```suggestion
```

File: src/Components/Login/Login.test.jsx [hunk 2]
```diff
@@ -1,9 +1,47 @@
...
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import userEvent from '@testing-library/user-event';
+import configureMockStore from 'redux-mock-store';
 import Login from './index';
+import Store from '../../Store';
 
 describe('Login', () => {
+  const MockStore = configureMockStore();
+  const store = Store;
```
Comment (pranavakkian) [inline src/Components/Login/Login.test.jsx]: Why u r using both mock store and real store?
```suggestion
```

File: src/Components/Login/Login.test.jsx [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
     expect(container).toMatchSnapshot();
   });
+  test('Login Failed', () => {
+    // userEvent.type(screen.getAllByRole('textbox')[0], '');
+    // userEvent.type(screen.getAllByRole('textbox')[1], '');
+    render(<Provider store={store}><Login /></Provider>);
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+    expect(screen.getByText('Invalid Credentials.')).toBeInTheDocument();
+    userEvent.click(screen.getByRole('button', { name: 'OK' }));
+    const { UserInfoReducer } = store.getState();
```
Comment (pranavakkian) [inline src/Components/Login/Login.test.jsx]: this is wrong
```suggestion
```

File: src/Components/Login/Login.test.jsx [hunk 4]
```diff
@@ -1,9 +1,47 @@
...
+        userName: 'test',
+      },
+    }));
+    const { container } = render(<Provider store={mockstore}><Login /></Provider>);
     expect(container).toMatchSnapshot();
   });
+  test('Login Failed', () => {
+    // userEvent.type(screen.getAllByRole('textbox')[0], '');
+    // userEvent.type(screen.getAllByRole('textbox')[1], '');
+    render(<Provider store={store}><Login /></Provider>);
```
Comment (pranavakkian) [inline src/Components/Login/Login.test.jsx:26]: use mock store

File: src/Components/Login/Login.test.jsx [hunk 5]
```diff
@@ -1,9 +1,47 @@
...
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+    expect(screen.getByText('Invalid Credentials.')).toBeInTheDocument();
+    userEvent.click(screen.getByRole('button', { name: 'OK' }));
+    const { UserInfoReducer } = store.getState();
+    expect(UserInfoReducer).toStrictEqual({ isLoggedIn: false, userName: null });
```
Comment (pranavakkian) [inline src/Components/Login/Login.test.jsx]:
```suggestion
    expect(store.getActions()).toStrictEqual({ ...expectedAtcion });
```

File: src/Components/Login/Login.test.jsx [hunk 6]
```diff
@@ -1,9 +1,47 @@
...
+    expect(UserInfoReducer).toStrictEqual({ isLoggedIn: true, userName: 'admin' });
+  });
+  test('Login Successful', () => {
+    render(<Provider store={store}><Login /></Provider>);
+    userEvent.type(screen.getAllByRole('textbox')[0], 'abc');
+    userEvent.type(screen.getAllByRole('textbox')[1], 'abc123');
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+    const { UserInfoReducer } = store.getState();
+    expect(UserInfoReducer).toStrictEqual({ isLoggedIn: true, userName: 'abc' });
+  });
```
Comment (pranavakkian) [inline src/Components/Login/Login.test.jsx]: same as above

File: src/Components/Login/LoginForm/index.jsx [hunk 7]
```diff
@@ -13,10 +15,13 @@ const LoginForm = () => {
         <Form.Field>
           <input placeholder="Password" value={password} label="Password" onChange={(e) => setPassword(e.target.value)} />
         </Form.Field>
-        <Button type="submit">Submit</Button>
+        <Button type="submit" onClick={() => { handleLoginForm(username, password); }}>Submit</Button>
```
Comment (pranavakkian) [inline src/Components/Login/LoginForm/index.jsx]:
```suggestion
        <Button type="submit" onClick={() => handleLoginForm(username, password)}>Submit</Button>
```

**Review by pranavakkian — CHANGES_REQUESTED** [review]

**Review by pranavakkian — APPROVED** [review]

**Review by shivalibaghel7 — APPROVED** [review]

**Pull request merged**